File: src/services/userService.js
import { api } from "./api";

export async function listUsers(params = {}) {
  return api.get("/users", { params }).then((r) => r.data);
}

export async function getUserById(id) {
  return api.get(`/users/${id}`).then((r) => r.data);
}

export async function createUser(payload) {
  return api.post("/users", payload).then((r) => r.data);
}

export async function updateUser(id, payload) {
  return api.put(`/users/${id}`, payload).then((r) => r.data);
}

export async function deleteUser(id) {
  return api.delete(`/users/${id}`).then((r) => r.data);
}

export async function updateStatus(id, status) {
  return api.patch(`/users/${id}/status`, { status }).then((r) => r.data);
}

export async function resetPassword(id, password, confirmPassword) {
  return api.patch(`/users/${id}/reset-password`, { password, confirmPassword }).then((r) => r.data);
}

export async function updateSelfProfile(payload) {
  return api.put("/users/profile", payload).then((r) => r.data);
}

File: backend/src/main/java/com/falconenergy/service/UserService.java
package com.falconenergy.service;

import com.falconenergy.dto.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface UserService {
    TokenResponse login(UserLoginRequest request);
    UserResponse register(UserRegisterRequest request);
    TokenResponse refreshToken(RefreshTokenRequest request);
    void logout(String token);
    UserResponse createUser(UserRegisterRequest request);
    UserResponse getUserById(Long id);
    UserResponse updateUser(Long id, UserUpdateRequest request);
    UserResponse updateStatus(Long id, StatusUpdateRequest request);
    UserResponse resetPassword(Long id, PasswordResetRequest request);
    UserResponse updateSelfProfile(String currentUserEmailOrUsername, SelfProfileUpdateRequest request);
    void deleteUser(Long id);
    Page<UserResponse> getAllUsers(String search, String role, String status, Pageable pageable);
}
